flink-connectors/flink-connector-cassandra/src/main/java/org/apache/flink/batch/connectors/cassandra/CassandraOutputFormatBase.java

```diff
@@ -86,4 +94,57 @@ public void close() throws IOException {
             }
         }
     }
+
+    protected static <T> CompletableFuture<T> listenableFutureToCompletableFuture(
+            final ListenableFuture<T> listenableFuture) {
+        CompletableFuture<T> completable =
+                new CompletableFuture<T>() {
```

Comment:
This whole implementation is a bit strange.

a) we dont need to worry about cancel because the base works against the CompletationStage interface which doesn't support cancellation.
b) get() references the listenableFuture, but when the listenableFuture completes we complete the future. This shouldn't work in both directions.

Just create a plain CompFuture, and keep the behavior in the callback.

Reply:
Agree